lib/providers/notifiers/connection_notifier.dart

```diff
@@ -13,31 +19,48 @@ class ConnectionNotifier extends ChangeNotifier {
     'https://mainnet.telosusa.io',
     'https://telos.eosphere.io',
     'https://telos.caleos.io',
+    'https://api.eos.miami',
+    'https://hyperion.telosgermany.io',
   ];
 
   void init() {
     discoverEndpoints();
   }
 
   void discoverEndpoints() async {
+    List<Future> checks = List<Future>();
+
     for (var endpoint in availableEndpoints) {
-      var ping = Stopwatch()..start();
+      checks.add(checkEndpoint(endpoint));
+    }
 
-      Response res = await get("$endpoint/v2/health");
+    var responses = await Future.wait(checks);
 
-      ping.stop();
+    responses.sort((a, b) => a.ping - b.ping);
 
-      if (res.statusCode == 200) {
-        int endpointPing = ping.elapsedMilliseconds;
+    //for (int i=0; i<responses.length; i++) {
+    //  print("$i ping: ${responses[i].ping} endpoint: "+responses[i].url);
+    //}
+    currentEndpoint = responses[0].url;
+    currentEndpointPing = responses[0].ping;
+    notifyListeners();
 
-        print("ping from $endpoint is $endpointPing");
+  }
 
-        if (currentEndpointPing == 0 || endpointPing < currentEndpointPing) {
-          currentEndpoint = endpoint;
-          currentEndpointPing = endpointPing;
-          notifyListeners();
-        }
+  Future<Endpoint> checkEndpoint(String endpoint) async {
+    try {
+      var ping = Stopwatch()..start();
+      Response res = await get("$endpoint/v2/health");
+      ping.stop();
+      if (res.statusCode == 200) {
+        int endpointPing = ping.elapsedMilliseconds;
+        return Endpoint(endpoint, endpointPing);
+      } else {
+        return Endpoint(endpoint, 1000000);
       }
+    } catch (err) {
+      print("error pinging: " + err);
+      return Endpoint(endpoint, 2000000);
```

Comment:
same here. 
Why 2000000 and not 1000000? 

These should be constants with meaningful names;